File: 283_Move_Zeroes.js
/**
 * Runtime: 136 ms
 */

/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    var l = nums.length;

    for (var i = 0; i < l; i++) {
        if (nums[i] === 0) {
            var z = nums.splice(i, 1);
            nums.push(z[0]);
            i--;
            l--;
        }
    }
};

File: 283_Move_Zeroes.c
void moveZeroes(int* nums, int numsSize) {
  int temp, i, len;
  len = numsSize;
  for (i = 0; i < numsSize; i++) {
    if (nums[i] == 0) {
      temp = nums[i];
      int j = i + 1;
      while(nums[j] == 0) { 
        j++; 
      }

      if (j <= len - 1) {
        nums[i] = nums[j];
        nums[j] = temp;
      } 
    }
  }
}